### DISCUSSION TITLE
Unable to install streamripper, mplayer (no bottle available)

### DISCUSSION THREAD

**Discussion by 1-om** [Homebrew/discussions#2714] [Casks]
I am experienced in linux, but new to Mac.

```
~/ brew doctor
Your system is ready to brew.
```
tried reinstalling brew, no effect
```
~/ brew install streamripper
Error: streamripper: no bottle available!
You can try to install from source with:
  brew install --build-from-source streamripper
Please note building from source is unsupported.
```
same for mplayer

when building bottles I get errors & I would like to avoid that approach

Thanks for help

**Comment by carlocab** [comment]
What is the output of `brew config`?

**Reply by 1-om** [reply]
~/ brew config
HOMEBREW_VERSION: 3.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96137bc19e68398ebbb7033379df288cd8b9a3f9
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9b79f47c8370c7449aa1470a0c3eb7386e8d5f3d
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: code -w
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false

**Reply by carlocab** [reply]
`streamripper` and `mplayer` don't build on ARM macOS. There's not much we can do about that, unfortunately. You could try building it from source yourself (`brew install --build-from-source --verbose`) and reporting the errors upstream to see if they can fix it.

**Reply by 1-om** [reply]
Thanks for looking into this.
So much so for the best machine from apple.

Any guide on how to report errors upstream please?

**Reply by carlocab** [reply]
That's project-dependent. A good way to start is to look at the homepage (e.g. `brew home streamripper`) and check if they have instructions for reporting bugs. They might have an issue tracker, forum, or mailing list for doing this.
